src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -218,46 +218,47 @@ namespace HeatBalFiniteDiffManager {
         // user settings for numerical parameters
         cCurrentModuleObject = "HeatBalanceSettings:ConductionFiniteDifference";
 
-        if (inputProcessor->getNumObjectsFound(state, cCurrentModuleObject) > 0) {
-            inputProcessor->getObjectItem(state,
-                                          cCurrentModuleObject,
-                                          1,
-                                          cAlphaArgs,
-                                          NumAlphas,
-                                          rNumericArgs,
-                                          NumNumbers,
-                                          IOStat,
-                                          lNumericFieldBlanks,
-                                          lAlphaFieldBlanks,
-                                          cAlphaFieldNames,
-                                          cNumericFieldNames);
-
-            if (!lAlphaFieldBlanks(1)) {
-
-                {
-                    auto const SELECT_CASE_var(cAlphaArgs(1));
-
-                    if (SELECT_CASE_var == "CRANKNICHOLSONSECONDORDER") {
-                        CondFDSchemeType = CrankNicholsonSecondOrder;
-                    } else if (SELECT_CASE_var == "FULLYIMPLICITFIRSTORDER") {
-                        CondFDSchemeType = FullyImplicitFirstOrder;
-                    } else {
-                        ShowSevereError(state, cCurrentModuleObject + ": invalid " + cAlphaFieldNames(1) + " entered=" + cAlphaArgs(1) +
+        auto instances = inputProcessor->epJSON.find(cCurrentModuleObject);
+        if (instances != inputProcessor->epJSON.end()) {
+            auto instance = instances.value().begin();
+            auto const &fields = instance.value();
+
+            if (fields.find("difference_scheme") != fields.end()) { // not required field, has default value
+                std::string scheme = fields.at("difference_scheme");
+                if (scheme == "CRANKNICHOLSONSECONDORDER") {
+                    CondFDSchemeType = CrankNicholsonSecondOrder;
+                } else if (scheme == "FULLYIMPLICITFIRSTORDER") {
+                    CondFDSchemeType = FullyImplicitFirstOrder;
+                } else {
+                    ShowSevereError(state,
+                                    cCurrentModuleObject + ": invalid Difference Scheme entered=" + scheme +
```

Comment:
getEnumValue, Difference Scheme is a choice field so else is never possible, which means format isn't needed because else can be deleted. 

    A1 , \field Difference Scheme
       \type choice
       \key CrankNicholsonSecondOrder
       \key FullyImplicitFirstOrder
       \default FullyImplicitFirstOrder

Reply:
This branch is behind develop and has obvious conflicts. This has already been converted to an enum.

    HeatBalFiniteDiffManager::CondFDScheme CondFDSchemeType =
        HeatBalFiniteDiffManager::CondFDScheme::FullyImplicitFirstOrder; // solution scheme for CondFD - default